src/controller/CHIPDeviceController.cpp

```diff
@@ -1201,6 +1215,41 @@ void DeviceCommissioner::OnOperationalCertificateSigningRequest(void * context,
     }
 }
 
+void DeviceCommissioner::OnDeviceNOCGenerated(void * context, const ByteSpan & noc, const PeerId & deviceId)
```

Comment:
deviceId is also not used here, suggest removing.

Reply:
`deviceId/nodeId` will be useful when we remove it from `PairDevice` API. The commissioner will need this to maintain the state of local/proxy device object list.

Reply:
But the actual node identity is the authenticated identity as defined by the tuple of (root CA public key, fabric ID, node ID) embedded within the signed certificate chain.  This identity may not have even been produced locally.

Why are we forcing the opCreds delegate to parse and extract the node ID from the NOC when this is not necessarily its core function?

The only thing that can happen by passing this duplicate information is that it could be incorrect and not match the certificate chain.  How do we properly guard against that?  It would seem this can only be guarded against by parsing the node ID RND here from the NOC and comparing.  At that point, why pass the node ID separately in the first place?  Again, this can only break and it's untestable because the delegate implementations are out-of-tree.

Reply:
This method is specific to device certificate. There's another callback for the NOC for the controller.

Reply:
@msandstedt I agree with your argument. Removing nodeId from the callbacks

Reply:
Marking as resolved